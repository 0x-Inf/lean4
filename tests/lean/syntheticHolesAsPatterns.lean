inductive Fam2 : Type → Type → Type 1 where
  | any : Fam2 α α
  | nat : Nat → Fam2 Nat Nat

example (a : α) (x : Fam2 α β) : β :=
  match α, β, a, x with
  | ?α, ?β, ?a, Fam2.any   => _
  | ?α, ?β, ?a, Fam2.nat n => _
