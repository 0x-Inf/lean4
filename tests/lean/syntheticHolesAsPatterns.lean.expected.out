syntheticHolesAsPatterns.lean:7:30-7:31: error: don't know how to synthesize placeholder
context:
α✝ β : Type
a✝ : α✝
x : Fam2 α✝ β
α : Type
a : α
⊢ α
syntheticHolesAsPatterns.lean:8:30-8:31: error: don't know how to synthesize placeholder
context:
α β : Type
a✝ : α
x : Fam2 α β
a n : Nat
⊢ Nat
